### PR TITLE
Bugfix quotacell validation

### PR DESCRIPTION
**Pull request by manikanta-talanki**

### CODE REVIEW

File: lib/validate.go [hunk 1]
```diff
@@ -135,20 +139,28 @@ func ValidateExclusionType(val ExclusionType) error {
...
+			return ErrMissingQuotaCells
 		}
-		for _, vc := range vq.QuotaCells {
+		var allocType Allocation
+		for i, vc := range vq.QuotaCells {
 			if vc.Perc == nil && vc.Count == nil {
-				return ErrInvalidQuotaCellValue
+				return ErrAllocationNotProvided
+			}
+			if vc.Perc != nil && vc.Count != nil {
```
Comment (ashajyothi828) [inline lib/validate.go]: Cant they send 0 for one of the values by default?

File: lib/validate_test.go [hunk 2]
```diff
@@ -81,3 +82,67 @@ func TestValidateSurveyLink(t *testing.T) {
...
+// TestValidateQuotaPlan testing quota plan validation
+func TestValidateQuotaPlan(t *testing.T) {
+	tables := []struct {
+		name     string
+		input    string
+		expected error
+	}{
+		{
+
+			"Case 1: Happy path, everything is valid with full quota plan",
```
Comment (ashajyothi828) [inline lib/validate_test.go]: Can we add a test case for when the both/ one of filters and quota groups are sent in as empty arrays?
